engine/classes/Elgg/WidgetsService.php

```diff
@@ -143,18 +143,21 @@ public function canEditLayout($context, $user_guid = 0) {
 			return false;
 		}
 
-		if ($user) {
-			$return = ($user->isAdmin() || (elgg_get_page_owner_guid() == $user->guid));
-		} else {
-			$return = false;
+		$page_owner = elgg_get_page_owner_entity();
+		$default = false;
+		
+		if ($page_owner) {
+			$default = $page_owner->canEdit($user_guid);
+		} elseif ($user) {
+			$default = $user->isAdmin();
```

Comment:
More efficient
```php
if ($user->isAdmin()) {
     $default = true;
} elseif ($page_owner) {
     $default = $page_owner->canEdit($user->guid);
}

Reply:
$user is not always available, so doing it this way